src/constants/chainInfo.ts

```diff
@@ -277,11 +277,15 @@ const CHAIN_INFO: ChainInfoMap = {
   },
 } as const
 
-export function getChainInfo(chainId: SupportedL1ChainId): L1ChainInfo
-export function getChainInfo(chainId: SupportedL2ChainId): L2ChainInfo
-export function getChainInfo(chainId: ChainId): L1ChainInfo | L2ChainInfo
+export function getChainInfo(chainId: SupportedL1ChainId, featureFlags?: Record<typeof chainId, boolean>): L1ChainInfo
+export function getChainInfo(chainId: SupportedL2ChainId, featureFlags?: Record<typeof chainId, boolean>): L2ChainInfo
 export function getChainInfo(
-  chainId: ChainId | SupportedL1ChainId | SupportedL2ChainId | number | undefined
+  chainId: ChainId,
+  featureFlags?: Record<typeof chainId, boolean>
+): L1ChainInfo | L2ChainInfo
+export function getChainInfo(
+  chainId: ChainId | SupportedL1ChainId | SupportedL2ChainId | number,
+  featureFlags?: Record<typeof chainId, boolean>
```

Comment:
this would allow you to pass in an undefined `chainId` as before which will help clean up several null checks required elswhere
```suggestion
  chainId: ChainId | SupportedL1ChainId | SupportedL2ChainId | number | undefined,
  featureFlags?: Record<ChainId | SupportedL1ChainId | SupportedL2ChainId | number, boolean>
```